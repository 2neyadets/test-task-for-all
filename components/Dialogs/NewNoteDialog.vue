<template lang="pug">
  Dialog(:value="newNoteDialog" @hide="changeDialogState" @keyup.esc.native="changeDialogState")
    NewNoteForm
</template>

<script>
import { mapGetters } from 'vuex'
import Dialog from '../Reusable/Dialog'
import NewNoteForm from '../Forms/NewNoteForm'

export default {
  name: 'NewNoteDialog',
  components: { Dialog, NewNoteForm },
  props: {},
  data () {
    return {
    }
  },
  computed: {
    ...mapGetters([
      'newNoteDialog',
    ]),
  },
  watch: {},
  created () {},
  mounted () {},
  updated () {},
  methods: {
    changeDialogState () {
      this.$store.dispatch('changeDialogState', 'newNote')
    },
  },
}
</script>
