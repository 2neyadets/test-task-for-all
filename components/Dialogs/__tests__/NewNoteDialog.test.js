import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import NewNoteDialog from '../NewNoteDialog'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('NewNoteDialog component', () => {
  let wrapper, actions, getters, store

  beforeEach(() => {
    actions = {
      changeDialogState: jest.fn(),
    }
    getters = {
      newNoteDialog: () => true,
    }
    store = new Vuex.Store({
      actions,
      getters,
    })

    wrapper = shallowMount(NewNoteDialog, {
      localVue,
      store,
    })
  })

  it('mounts without errors', done => {
    localVue.config.errorHandler = done // для отлавливания ошибок внутри nextTick, если нужно что-то выпонять внутри него
    expect.assertions(1)
    expect(wrapper).toBeTruthy()
    done()
  })

  it('can handle hide event', async () => {
    expect.assertions(1)
    wrapper.find('NewNoteForm-stub').vm.$parent.$emit('hide')
    await localVue.nextTick()
    expect(actions.changeDialogState).toHaveBeenCalled()
  })

  it('match snapshot', () => {
    expect.assertions(1)
    expect(wrapper.element).toMatchSnapshot()
  })
})
