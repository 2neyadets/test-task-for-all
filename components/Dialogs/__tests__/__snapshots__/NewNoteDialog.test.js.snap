// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewNoteDialog component match snapshot 1`] = `
<dialog-stub
  value="true"
>
  <newnoteform-stub />
</dialog-stub>
`;
